bin/reflective_loader/src/loader.rs

```diff
@@ -0,0 +1,844 @@
+
+    fn c_size() -> usize {
+        core::mem::size_of::<u16>()
+    }
+}
+
+#[cfg(target_arch = "x86_64")]
+struct PeFileHeaders64 {
+    dos_headers: IMAGE_DOS_HEADER,
+    nt_headers: IMAGE_NT_HEADERS64,
+    section_headers: [IMAGE_SECTION_HEADER; MAX_PE_SECTIONS],
+}
+
+// Pares the PE file from a series of bytes
+#[cfg(target_arch = "x86_64")]
+impl PeFileHeaders64 {
+    fn new(dll_bytes_ptr: *mut c_void) -> Self{
+        // DOS Headers
+        let dos_headers_base_ptr = dll_bytes_ptr as usize;
+        let dos_headers = unsafe { *((dos_headers_base_ptr) as *mut IMAGE_DOS_HEADER) };
+        if dos_headers.e_magic != PE_MAGIC {
+            panic!("PE Magic header mismatch. Expected 0x5A4D == MZ == 21117. File does not appear to be a PE executable.");
+        }
+
+        // NT Headers
+        let nt_headers_base_ptr = dos_headers_base_ptr + dos_headers.e_lfanew as usize;
+        let nt_headers = unsafe { *((nt_headers_base_ptr) as *mut IMAGE_NT_HEADERS64) };
+        if nt_headers.Signature != NT_SIGNATURE {
+            panic!("NT Signature mismatch. Expected 0x4550 == PE == 17744. File does not appear to be a PE executable.");
+        }
+
+        // Section Headers - hopefully there isn't more than MAX_PE_SECTIONS sections.
+        let null_section = IMAGE_SECTION_HEADER{ 
+            Name: [0; 8], 
+            Misc: IMAGE_SECTION_HEADER_0 { 
+                PhysicalAddress: 0, 
+            },
+            VirtualAddress: 0, 
+            SizeOfRawData: 0, 
+            PointerToRawData: 0, 
+            PointerToRelocations: 0, 
+            PointerToLinenumbers: 0, 
+            NumberOfRelocations: 0, 
+            NumberOfLinenumbers: 0, 
+            Characteristics: 0
+        };
+        let mut section_headers: [IMAGE_SECTION_HEADER; MAX_PE_SECTIONS] = [null_section; MAX_PE_SECTIONS];
+        let optional_headers_start_ptr = unsafe{&(*(nt_headers_base_ptr as *mut IMAGE_NT_HEADERS64)).OptionalHeader as *const _ as usize};
+        let section_headers_start_ptr = optional_headers_start_ptr + nt_headers.FileHeader.SizeOfOptionalHeader as usize;
+        let mut cur_section_ptr = section_headers_start_ptr as *mut IMAGE_SECTION_HEADER;
+        for section_index in 0..nt_headers.FileHeader.NumberOfSections {
+            let cur_section = unsafe { *cur_section_ptr.clone() };
+
+            section_headers[section_index as usize] = cur_section;
+
+            cur_section_ptr =
+                    (cur_section_ptr as usize + core::mem::size_of::<IMAGE_SECTION_HEADER>() as usize) as *mut IMAGE_SECTION_HEADER 
+        }    
+
+        Self {
+            dos_headers,
+            nt_headers,
+            section_headers,
+        }
+    }
+}
+
+/// In order to keep the compiler from being upset about
+/// not being able to find memmove or memcpy we need to 
+/// implement our own copy function that doesn't call etiher.
+#[no_mangle]
+pub unsafe extern "C" fn memcpy(dest: *mut u8, src: *const u8, n: usize) -> *mut u8 {
```

Comment:
You may just be able to use `compiler-builtins-mem` if you are using `build-std`

Reply:
Oh that's fire. 🔥

Reply:
@wumb0 
I'm having a hard time getting this to work.
I added 
<img width="474" alt="image" src="https://github.com/KCarretto/realm/assets/7121375/88dcff39-d8a7-4a37-a157-50ca84293b9e">
<img width="1527" alt="image" src="https://github.com/KCarretto/realm/assets/7121375/a78c7259-1628-408a-9c97-f03dd27f62a2">

Reply:
Try passing the rustflag -Z share-generics=n and define a _fltused next to your panic function in lib.rs
```rust
extern "C" int _fltused = 0;
```

also, don't use memcpy, use std::mem::copy_nonoverlapping.
be careful though, its src and dest args are swapped from memcpy.

Reply:
GOT IT! Working 🎉